torchaudio/prototype/pipelines/hifigan_pipeline.py

```diff
@@ -0,0 +1,224 @@
+            f_min: Optional[float],
+            f_max: Optional[float],
+            sample_rate: float,
+            n_mels: int,
+        ):
+            super(HiFiGANVocoderBundle._HiFiGANMelSpectrogram, self).__init__()
+            self.mel_transform = MelSpectrogram(
+                sample_rate=sample_rate,
+                n_fft=n_fft,
+                win_length=win_length,
+                hop_length=hop_size,
+                f_min=f_min,
+                f_max=f_max,
+                n_mels=n_mels,
+                normalized=False,
+                pad=0,
+                mel_scale="slaney",
+                norm="slaney",
+                center=False,
+            )
+            self.sample_rate = sample_rate
+            self.hop_size = hop_size
+            self.n_fft = n_fft
+            self.win_length = win_length
+            self.f_min = f_min
+            self.f_max = f_max
+            self.n_mels = n_mels
+            self.pad_size = int((n_fft - hop_size) / 2)
+
+        def forward(self, waveform: torch.Tensor) -> torch.Tensor:
+            """Generate mel spectrogram from a waveform. Should have same sample rate as ``self.sample_rate``.
+
+            Args:
+                waveform (Tensor): waveform of shape ``(batch_size, time_length)``.
+            Returns:
+                Tensor of shape ``(batch_size, n_mel, time_length)``
+            """
+            ref_waveform = F.pad(waveform.unsqueeze(1), (self.pad_size, self.pad_size), mode="reflect")
+            ref_waveform = ref_waveform.squeeze(1)
+
+            spectr = (self.mel_transform.spectrogram(ref_waveform) + 1e-9) ** 0.5
+            mel_spectrogram = self.mel_transform.mel_scale(spectr)
+            mel_spectrogram = torch.log(torch.clamp(mel_spectrogram, min=1e-5))
+            return mel_spectrogram
+
+
+HIFIGAN_VOCODER_V3_LJSPEECH = HiFiGANVocoderBundle(
+    "hifigan_generator_v3_ljspeech.pth",
+    _vocoder_params={
+        "upsample_rates": (8, 8, 4),
+        "upsample_kernel_sizes": (16, 16, 8),
+        "upsample_initial_channel": 256,
+        "resblock_kernel_sizes": (3, 5, 7),
+        "resblock_dilation_sizes": ((1, 2), (2, 6), (3, 12)),
+        "resblock_type": 2,
+        "in_channels": 80,
+        "lrelu_slope": 0.1,
+    },
+    _mel_params={
+        "hop_size": 256,
+        "n_fft": 1024,
+        "win_length": 1024,
+        "f_min": 0,
+        "f_max": 8000,
+    },
+    _sample_rate=22050,
+)
+HIFIGAN_VOCODER_V3_LJSPEECH.__doc__ = """Pre-trained HiFiGAN Vocoder pipeline, transforming mel spectrograms into
+    waveforms. The underlying model is constructed by :py:func:`torchaudio.prototype.models.hifigan_generator`
+    and utilizes weights trained on *The LJ Speech Dataset* :cite:`ljspeech17`. The weights are converted from the ones
+    published with the original paper :cite:`NEURIPS2020_c5d73680` (See links to pre-trained models on
+    `GitHub <https://github.com/jik876/hifi-gan#pretrained-model>`__).
```

Comment:
> The weights are converted from the one published with the original paper

Can you also mention and link the license of the original model?

Reply:
Linked MIT licence from the original repo